scripts/lib/k8s.sh

```diff
@@ -1,25 +1,59 @@
 #!/usr/bin/env bash
 
+CONTROLLER_TOOLS_VERSION="v0.4.0"
+
+# ensure_controller_gen checks that the `controller-gen` binary is available on
+# the host system and if it is, that it matches the exact version that we
+# require in order to standardize the YAML manifests for CRDs and Kubernetes
+# Roles.
+#
+# If the locally-installed controller-gen does not match the required version,
+# prints an error message asking the user to uninstall it.
 ensure_controller_gen() {
-  if ! is_installed controller-gen; then
-    # Need this version of controller-gen to allow dangerous types and float
-    # type support
-    go get sigs.k8s.io/controller-tools/cmd/controller-gen@4a903ddb7005459a7baf4777c67244a74c91083d
-  else
-    minimum_req_version="v0.3.1"
-    # Don't overide the existing version let the user decide.
-    if ! is_min_controller_gen_version "$minimum_req_version"; then
-        echo "Existing version of controller-gen "`controller-gen --version`", minimum required is $minimum_req_version"
-        exit 1
+    if ! is_installed controller-gen; then
+        __install_path=/usr/local/bin/controller-gen
+        __work_dir=$(mktemp --tmpdir -d controller-gen-XXX)
```

Comment:
I can confirm this script is working on Debian-based Linux ditributions at least, however `mktemp` on Darwin works differently 😕 . Basically on darwin, `mktemp` doesn't accept a `--tmpdir` flag along with small other differences. 

Instead of dealing with each case separately, we can just use the common `-d` flag, and hard code (or use a `$TMPDIR` environment variable) the temp directory.
Using `mktemp -d /tmp/controller-gen-XXX` seems to be the right/fairest solutions for me.